core/db/migrate/20210921070813_create_spree_wishlists.rb

```diff
@@ -0,0 +1,18 @@
+class CreateSpreeWishlists < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_wishlists, if_not_exists: true do |t|
+      t.belongs_to :user
+      t.belongs_to :store
+
+      t.column :name, :string
+      t.column :token, :string, null: false
+      t.column :is_private, :boolean, default: true, null: false
+      t.column :is_default, :boolean, default: false, null: false
+
+      t.timestamps
+    end
+
+    add_index :spree_wishlists, :token, unique: true
```

Comment:
@MatthewKennedy this `token` column didn't exist in the old wishlist gem right? It was `access_hash` so if the table exists let's rename `access_hash` to `token` so it will be possible to migrate over

Reply:
@damianlegawiec the migration before this one renames `access_hash` to `token` I'll do another  check now by installing the old gem on a copy of 4.3 and then migrating to this version and check the schema difference...

Reply:
Schema After Installing fresh 4.3 and Spree_Wishlists Ext
```ruby
  create_table "spree_wished_products", id: :serial, force: :cascade do |t|
    t.bigint "variant_id"
    t.bigint "wishlist_id"
    t.text "remark"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.integer "quantity", default: 1, null: false
  end

  create_table "spree_wishlists", id: :serial, force: :cascade do |t|
    t.bigint "user_id"
    t.string "name"
    t.string "access_hash"
    t.boolean "is_private", default: true, null: false
    t.boolean "is_default", default: false, null: false
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.index ["user_id", "is_default"], name: "index_spree_wishlists_on_user_id_and_is_default"
    t.index ["user_id"], name: "index_spree_wishlists_on_user_id"
  end
```

Schema After UpgradingThe Same Project Above To This PR

```ruby
  create_table "spree_wished_items", id: :serial, force: :cascade do |t|
    t.bigint "variant_id"
    t.bigint "wishlist_id"
    t.text "remark"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.integer "quantity", default: 1, null: false
    t.index ["variant_id", "wishlist_id"], name: "index_spree_wished_items_on_variant_id_and_wishlist_id", unique: true
    t.index ["variant_id"], name: "index_spree_wished_items_on_variant_id"
    t.index ["wishlist_id"], name: "index_spree_wished_items_on_wishlist_id"
  end

  create_table "spree_wishlists", id: :serial, force: :cascade do |t|
    t.bigint "user_id"
    t.string "name"
    t.string "token"
    t.boolean "is_private", default: true, null: false
    t.boolean "is_default", default: false, null: false
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.bigint "store_id"
    t.index ["store_id"], name: "index_spree_wishlists_on_store_id"
    t.index ["token"], name: "index_spree_wishlists_on_token", unique: true
    t.index ["user_id", "is_default"], name: "index_spree_wishlists_on_user_id_and_is_default"
    t.index ["user_id"], name: "index_spree_wishlists_on_user_id"
  end
```

--------

Schema From Project That Was Created On This PR
```ruby
  create_table "spree_wished_items", force: :cascade do |t|
    t.bigint "variant_id"
    t.bigint "wishlist_id"
    t.integer "quantity", default: 1, null: false
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.index ["variant_id", "wishlist_id"], name: "index_spree_wished_items_on_variant_id_and_wishlist_id", unique: true
    t.index ["variant_id"], name: "index_spree_wished_items_on_variant_id"
    t.index ["wishlist_id"], name: "index_spree_wished_items_on_wishlist_id"
  end

  create_table "spree_wishlists", force: :cascade do |t|
    t.bigint "user_id"
    t.bigint "store_id"
    t.string "name"
    t.string "token", null: false
    t.boolean "is_private", default: true, null: false
    t.boolean "is_default", default: false, null: false
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.index ["store_id"], name: "index_spree_wishlists_on_store_id"
    t.index ["token"], name: "index_spree_wishlists_on_token", unique: true
    t.index ["user_id", "is_default"], name: "index_spree_wishlists_on_user_id_and_is_default"
    t.index ["user_id"], name: "index_spree_wishlists_on_user_id"
  end
```

Reply:
Looks good to me, the `remark` column is left in so if someone has been using it we are not dropping any of their data.